pyspeckit/spectrum/plotters.py

```diff
@@ -378,14 +402,13 @@ def label(self, title=None, xlabel=None, ylabel=None, verbose_label=False,
 
         if xlabel is not None:
             self.xlabel = xlabel
-        elif isinstance(self.Spectrum.xarr.xtype,str):
-            self.xlabel = self.Spectrum.xarr.xtype.title()
+        elif self._xunit:
+            self.xlabel += " ("+u.Unit(self._xunit).to_string()+")"
             if verbose_label:
                 self.xlabel = "%s %s %s" % ( self.Spectrum.xarr.velocity_convention.title(),
-                        self.Spectrum.xarr.frame.title(),
+                        # self.Spectrum.xarr.frame.title(),
```

Comment:
If you remove this, you need to remove one of the `%s` formatting strings.  Is there any way to recover the `frame` from the unit?  I think the answer is "no", so removing it is the right thing to do, at least for now.

(physics lesson)
The `frame` means the rest frame of the data.  So for example, there is the "topocentric" frame, which means the frame of the observatory.  "geocentric" means the Earth's center of mass frame - "topocentric" is moving at up to 0.5 km/s relative to "geocentric" because of the Earth's rotation.  Then there are "heliocentric" (sun-centered) and "barycentric" (around the center of mass of our solar system, which is tilted a little bit toward Jupiter), and then "Local Standard of Rest", which is defined by the average motions of stars near Earth.